### PR TITLE
Added xSQLAlias resource

### PR DESCRIPTION
**Pull request by Matticusau**
Have added the xSqlAlias resource which I developed for xSqlPs but am
merging my changes into xSqlServer as it seems to be more actively
maintained.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xsqlserver/23)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,6 +202,14 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
...
+
+* **SQLServerName**: The name of Alias (e.g. svr01\inst01).
+* **ServerName**: The name of real server.
+* **Protocol**: The protocol of either tcp or np (named pipes).
+* **RetryCount**: Maximum number of retries to check HA group existency.
+* **TCPPort**: The tcp port of the instance.
+
 ## Versions
 
 <<<<<<< HEAD
```
Comment (TravisEz13) [inline README.md]: You should also add a section like this

## Unreleased
- Resources added
  - xSqlAliase

**Comment by TravisEz13** [comment]
It is required that you provide adequate test coverage for the code you change. [Contribution Guidelines -Adding test coverage for DSC resources](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#adding-test-coverage-for-dsc-resources)  Please work with us to come up with a plan to cover this code.

**Comment by Matticusau** [comment]
Hey @TravisEz13 i am confused because in my other contribution you mentioned i needed an empty line at the end of the file yet Appveyor fails the test when that exists?

**Comment by TravisEz13** [comment]
This failed because the code cannot merge, not the new line at the end of the file.

**Comment by TravisEz13** [comment]
@Matticusau Is this PR duplicate?

**Comment by Matticusau** [comment]
No this is the same resource that I contributed to xSqlPs but as there is no functionality for this in xSqlServer either I thought I would share the code between them (also seeing as it seemed like I would never get it merged into xSqlPs). There is also some community discussions around the fact that we have two SQL resources, what are your thoughts?

**Comment by TravisEz13** [comment]
As both resources already exist, they should be maintained in parity, or one should be deprecated, the other taking over the functionality (with a warning added to  start using the other.)  If we maintain both, I would suggest creating a 3rd common PS module which both package and use to implement the functionality.

**Comment by TravisEz13** [comment]
Please resolve all merge conflicts.  For more information see [GitHowTo - 30. Resolving Conflicts](http://githowto.com/resolving_conflicts )
[Instructions to setup BC4, my favorite merge tool](http://www.scootersoftware.com/support.php?zz=kb_vcs#gitwindows)

If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.  Please consider closing and re-opening when ready for review.

**Comment by Matticusau** [comment]
Hi @TravisEz13 thanks for the message, I will work to have this resolved today/tomorrow. As you know we have had other internal discussions around all of this so I will review the current pull request in line with those discussions to make sure we are heading in the right direction.

**Comment by Matticusau** [comment]
Hi @TravisEz13 thanks for the message, I will work to have this resolved today/tomorrow. As you know we have had other internal discussions around all of this so I will review the current pull request in line with those discussions to make sure we are heading in the right direction.

File: Tests/MSFT_xSqlAlias.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    It 'Should return true when Test is passed as Alias thats already set'{
+        Test-xSqlAliasTargetResource -SQLServerName 'myServerAlias'  -Protocol 'TCP' -ServerName localhost -TCPPort 1433 -Ensure 'Present' | Should Be $true
+    }
+
+    It 'Should return false when Test is passed as Alias that is not set'{
+        Test-xSqlAliasTargetResource -SQLServerName 'myServerAlias'  -Protocol 'TCP' -ServerName localhost -TCPPort 52002 -Ensure 'Present' | Should Be $false
+    }
+
+}
```
Comment (TravisEz13) [inline Tests/MSFT_xSqlAlias.Tests.ps1]: Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

File: README.md [hunk 3]
```diff
@@ -23,8 +23,13 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 * **xSQLServerMaxDop** resource to manage MaxDegree of Parallism for SQL Server
 * **xSQLServerMemory** resource to manage Memory for SQL Server
 * **xSQLServerPowerPlan** resource to manage windows powerplan on SQL Server
+<<<<<<< HEAD
```
Comment (TravisEz13) [inline README.md]: merge error +<<<<<<< HEAD should not be in the file   look for >>>>>> lines as well

File: README.md [hunk 4]
```diff
@@ -222,6 +238,7 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 ### 1.4.0.0
 
 * Resources Added
+  - xSqlAlias
```
Comment (TravisEz13) [inline README.md]: please move this to unreleased

File: Tests/MSFT_xSqlAlias.Tests.ps1 (hunk 2 above)
Comment (TravisEz13) [inline Tests/MSFT_xSqlAlias.Tests.ps1]: I would not block for this if it was the only issue, but please try to follow the style guidelines.

**Comment by TravisEz13** [comment]
A couple of blocking issues in the comments.  I didn't see any other major issues.

File: README.md [hunk 5]
```diff
@@ -202,6 +207,16 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 ###xSQLServerPowerPlan
 * **Ensure**: (key) An enumerated value that describes if Min and Max memory is configured
 
+<<<<<<< HEAD
+### xSqlAlias
+
+* **SQLServerName**: The name of Alias (e.g. svr01\inst01).
```
Comment (TravisEz13) [inline README.md]: perhaps the name of this could be clearer.  @HemantMahawar  do You have any suggestions?

File: README.md [hunk 6]
```diff
@@ -202,6 +207,16 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 ###xSQLServerPowerPlan
 * **Ensure**: (key) An enumerated value that describes if Min and Max memory is configured
 
+<<<<<<< HEAD
```
Comment (TravisEz13) [inline README.md]: merge error

File: README.md [hunk 7]
```diff
@@ -23,8 +23,13 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 * **xSQLServerMaxDop** resource to manage MaxDegree of Parallism for SQL Server
 * **xSQLServerMemory** resource to manage Memory for SQL Server
 * **xSQLServerPowerPlan** resource to manage windows powerplan on SQL Server
+<<<<<<< HEAD
+* **xSQLAlias** resource to manage SQL Server client Aliases
+
+=======
 * **xSQLServerNetwork** resource to manage SQL Server Network Protocols
 * **xSQLServerDatabase** resource to manage ensure database is present or absent
+>>>>>>> refs/remotes/PowerShell/dev
```
Comment (TravisEz13) [inline README.md]: merge error

File: README.md [hunk 8]
```diff
@@ -215,6 +230,7 @@ Please check out common DSC Resources [contributing guidelines](https://github.c
 * **Ensure**: An enumerated value that describes if Database is to be present or absent.
 * **SQLServer**: The SQL Server for the database
 * **SQLInstance**: The SQL instance for the database 
+>>>>>>> refs/remotes/PowerShell/dev
```
Comment (TravisEz13) [inline README.md]: merge error

**Comment by TravisEz13** [comment]
If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.  Please consider closing and re-opening when ready for review.

**Comment by Matticusau** [comment]
@travisEz13 I am going through these conflicts you have mentioned this weekend

**Comment by TravisEz13** [comment]
Reviewed 1 of 4 files at r1, 2 of 2 files at r2, 1 of 1 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/23)*

File: README.md (hunk 5 above)
Comment (Matticusau) [inline README.md]: After thinking about this I agree and have updated this to be Name in my dev branch. This adheres better to the standards of resource design and makes sense in the context.

**Comment by TravisEz13** [comment]
Reviewed 4 of 4 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/23)*

**Comment by mbreakey3** [comment]
@Matticusau: Any updates on this PR? If not, I'm going to close it - if there are any new updates it can be reopened

**Comment by kwirkykat** [comment]
These changes are now included in #89

**Pull request closed without merging**